design-proposals/virtiofs-support/virtiofs-feature-gates.md

```diff
@@ -0,0 +1,71 @@
+# Overview
+Virtiofs allows sharing Secrets, ConfigMaps, ServiceAccounts and DownwardAPI (a.k.a., ConfigVolumes) in a dynamic way,
+i.e., any change in those objects are reflected in the VirtualMachine (VM) without restarting the VM. Moreover, PVCs may
+also be shared using virtiofs allowing multiple readers and writers in the shared PVCs.
+
+## Motivation
+Currently, the usage of feature gates are mixed with the feature control. The feature gate `ExperimentalVirtiofsSupport`
+controls whether virtiofsd runs as rootful container inside the launcher pod or not. The feature gate ExperimentalVirtiofsSupport
+controls whether virtiofs can be used, and it automatically runs as rootful container inside the launcher pod when
+needed. There is no way to only run virtiofs for use cases which doesn't require rootful virtiofsd or the other way
+around.
+
+In order to GA virtiofs, VMs should be able to live migrate while sharing data with virtiofs. The virtiofsd’s live
+migration support will distinguish between different use cases:  sharing ConfigVolumes and PVCs. 
+
+In order to fully support sharing ConfigVolumes and PVCs while allowing live migration, deep changes are required in
+KubeVirt, QEMU, libvirt and virtiofsd projects. However, those changes are complementary, i.e., the changes required to
+support sharing ConfigVolumes while allowing a VM to live migrate are the needed and preliminary steps to support
+sharing PVCs. Therefore, this distinction will create a staggered and granular support giving users time to test the
+ConfigVolumes sharing support and allowing the involved developers to get feedback before next steps to support sharing
+PVCs.
+
+## Goals
+- Align virtiofs feature gates to just control the feature enablement.
+- Deprecate current virtiofs feature gate.
+- Protect the sharing functionalities: ConfigVolumes and PVCs, with feature gates.
+
+## Non Goals
+- Support live migration while sharing data with virtiofs.
+
+## Definition of Users
+A VM owner with permissions to read ConfigVolumes and/or read and/or write PVCs.
+
+## User Stories
+As a VM owner, I want to share a ConfigVolume in a dynamic way. For this purpose,
```

Comment:
A VM owner shouldn't have access to feature gates. We should rephrase to something like, As an admin I want to allow VM owners to share... without running root.

Reply:
Done.